controllers/imagerepository_notifications.go

```diff
@@ -0,0 +1,173 @@
+					updatedNotification, err := r.QuayClient.UpdateNotification(
+						r.QuayOrganization,
+						imageRepository.Spec.Image.Name,
+						statusNotification.UUID,
+						quay.Notification{
+							Title:  notification.Title,
+							Event:  string(notification.Event),
+							Method: string(notification.Method),
+							Config: quay.NotificationConfig{
+								Url: notification.Config.Url,
+							},
+							EventConfig: quay.NotificationEventConfig{},
+						})
+					if err != nil {
+						log.Error(err, "failed to update notification", "Title", statusNotification.Title, "UUID", statusNotification.UUID)
+						if err = r.UpdateImageRepositoryStatusMessage(ctx, imageRepository, "Error while updating the notification in Quay"); err != nil {
+							return err
+						}
+						return nil
+					}
+					statusNotification.UUID = updatedNotification.UUID
+				}
+			} else {
+				quayNotification := notificationExistsInQuayByTitle(notification.Title, imageRepository, allNotifications)
+				if quayNotification.UUID == "" {
+					// New item in Spec.Notifications: add notification to Quay and Status.Notifications
+					log.Info("Adding new notification to Quay", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method)
+					notificationStatus, err := r.AddNotification(notification, imageRepository)
+					if err != nil {
+						log.Error(err, "failed to add notification", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method)
+						if err = r.UpdateImageRepositoryStatusMessage(ctx, imageRepository, "Error while adding a notification to Quay"); err != nil {
+							return err
+						}
+						return nil
+					}
+					imageRepository.Status.Notifications = append(imageRepository.Status.Notifications, notificationStatus)
+				}
+			}
+		}
+		if existsInSpec == false {
+			// Deleted item from Spec.Notifications: delete notification from Quay and Status.Notifications
+			log.Info("Deleting notification in Quay", "Title", statusNotification.Title, "UUID", statusNotification.UUID)
+			_, err := r.QuayClient.DeleteNotification(
+				r.QuayOrganization,
+				imageRepository.Spec.Image.Name,
+				statusNotification.UUID)
+			if err != nil {
+				log.Error(err, "failed to delete notification", "Title", statusNotification.Title, "UUID", statusNotification.UUID)
+				if err = r.UpdateImageRepositoryStatusMessage(ctx, imageRepository, "Error while deleting the notification from Quay"); err != nil {
+					return err
+				}
+				return nil
+			}
+			// Remove notification from CR status
+			imageRepository.Status.Notifications = append(imageRepository.Status.Notifications[:index], imageRepository.Status.Notifications[index+1:]...)
+		}
+	}
+
+	return nil
+}
+
+func (r *ImageRepositoryReconciler) AddNotifications(ctx context.Context, imageRepository *imagerepositoryv1alpha1.ImageRepository) ([]imagerepositoryv1alpha1.NotificationStatus, error) {
```

Comment:
It overwrites all existing notifications in status. So, the name is confusing. Please rename or/and add a comment that explains the behavior of this method.

Reply:
Maybe we should raname it to `SetNotifications`?